File: 20180714/Maximum Average SubarrayI.py
class Solution(object):
    def findMaxAverage(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: float
        """
        if k > len(nums):
            return sum(nums) / len(nums)
        else:
            cur_sum = 0
            for i in range(k):
                cur_sum += nums[i]
            res = cur_sum
            for i in range(1,len(nums) - k + 1):
                cur_sum = cur_sum - nums[i - 1] + nums[i + k - 1]
                res = max(res, cur_sum)
            return res * 1.0 / k

File: 20180714/Maximum Average SubarrayI.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        if(k > nums.length){
            int sum = 0;
            for(int i :nums){
                sum += i;
            }
            return sum / nums.length;
        }
        else {
            double cur_sum = 0;
            for(int i = 0; i < k; i ++){
                cur_sum += nums[i];
            }
            double res = cur_sum;
            for(int j = 1; j < nums.length - k + 1; j ++){
                cur_sum = cur_sum  - nums[j-1] + nums[j + k - 1];
                res = Math.max(res, cur_sum);
            }
            return res / k;
        }
    }
}
